tests/test_compile_error.rs

```diff
@@ -33,7 +33,7 @@ fn test_compile_errors() {
     t.compile_fail("tests/ui/invalid_pymethod_receiver.rs");
     t.compile_fail("tests/ui/missing_intopy.rs");
     // adding extra error conversion impls changes the output
-    #[cfg(all(target_os = "linux", not(any(feature = "eyre", feature = "anyhow"))))]
```

Comment:
Why this modification?

Reply:
It's a bit of a drive-by but I was working on macOS today and I broke CI in the first attempt because this UI test regressed. Turns out the macos and linux outputs are the same.

Reply:
(I think) I added this because the output is different on Windows. The error message for this lists all the variants, and windows has additional error classes.

Reply:
Yep that's what I recall too, which is the why I left windows in the `not(any())` list